agents/demo/demo.ipynb

```diff
@@ -0,0 +1,970 @@
+   "metadata": {},
+   "source": [
+    "## Agents on Kubeflow 🤓"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this tutorial we will be training a reinforcement learning agent from the [tensorflow/agents](https://github.com/tensorflow/agents) project on Kubernetes using [Kubeflow](https://github.com/google/kubeflow)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The task the agent will be learning to perform is to operate a Kuka Robotics arm simulated in the OpenAI Gym Bullet Physics 'KukaBulletEnv-v0' environment. Feel free to [skip to the end](http://localhost:8888/notebooks/kubeflow-rl/apps/agents_ppo/demo.ipynb#Rendering-the-model) to see what this will look like!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Setup"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We need to create a Google Cloud Storage bucket to store job logs as well as a unique subdirectory of that bucket to store logs for this particular run. With the following we first create the GCS bucket then generate the path of a log dir to use in a later step."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Set the variables below to a project and bucket suitable for your use."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# GCP project to use\n",
+    "PROJECT=\"kubeflow-rl\"\n",
+    "\n",
+    "# Bucket to use\n",
+    "BUCKET=PROJECT+\"-kf\"\n",
+    "\n",
+    "# K8s cluster to use\n",
+    "CLUSTER=\"kubeflow-5fc52116\"\n",
+    "ZONE=\"us-east1-d\"\n",
+    "NAMESPACE=\"rl\"\n",
+    "\n",
+    "# Find the path to the workspace ml-app dir \n",
+    "import subprocess, os\n",
+    "def find_workspace_root(cwd=None):\n",
+    "    if cwd is None:\n",
+    "        cwd = subprocess.check_output([\"pwd\"]).strip()\n",
+    "    files = subprocess.check_output([\"ls\", cwd])\n",
+    "    if \"app.yaml\" in files:\n",
+    "        return cwd\n",
+    "    else:\n",
+    "        cwd = '/'.join(os.path.split(cwd)[:-1])\n",
+    "        return find_workspace_root(cwd)\n",
+    "APP_ROOT=find_workspace_root()\n",
+    "\n",
+    "# Needed for launching tensorboard\n",
+    "SECRET_NAME = \"gcp-credentials\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Attention:** You will need GCP credentials to access the cluster and GCP resources.\n",
+    "\n",
+    "If you're running this on your local machine you can authenticate for the GCP project you specified above in the usual way (i.e. `gcloud auth login` followed by `gcloud config set project <your-project-name>`; in this case skip the following step.\n",
+    "\n",
+    "If you're running this on JupyterLab you can do the following to provide the right credentials:\n",
```

Comment:
We should expand on this - either another notebook or the README solution I proposed above.